edgexfoundry/edgex-go
# feat: Implement service metrics for Events and Readings persisted

closes #3901

Related dependent PRs: 
- https://github.com/edgexfoundry/go-mod-core-contracts/pull/709
- bootstrap TBD

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I have opened a PR for the related docs change (if not, why?) **TBD on agreement on approach**
  <link to docs PR>

## Testing Instructions
- Run non-secure EdgeX stack with MQTT as the MessageBus and stop core data
  - `make run no-secty ds-virtual mqtt-bus` 
- Clone this branch, Build and run Core Data with DEBUG logging on
  - Also clone related PR branches for go-mod-core-contracts and go-mod-boostrap
- Run MQTT.fx , connect to broker on `localhost` and subscribe to `edgex/telemetry/#` topic
- When data received, copy the Base64 encoded payload and paste it in Decoder tool (https://www.base64decode.org/)
- Verify payload is a Metric DTO for `EventsPersisted` or `ReadingsPersisted`
- Repeat for next message received and verify it is the other Metric DTO.

## New Dependency Instructions (If applicable)
**N/A**
